test/activities/CategoriesEntity.js

```diff
@@ -0,0 +1,88 @@
+/* global fetchMock */
+
+import { CategoriesEntity } from '../../src/activities/CategoriesEntity.js';
+import { nonEditableCategories } from './data/NonEdtiableCategories.js';
+import { editableCategories } from './data/EditableCategories.js';
+import { emptyCategories } from './data/EmptyCategories.js';
+
+const expectedCategory = {
+	name: 'category1',
+	selected: true,
+	index: 0,
+};
+
+describe('CategoriesEntity', () => {
+	var editableEntity, nonEditableEntity, emptyEntity;
+
+	beforeEach(() => {
+		nonEditableEntity = window.D2L.Hypermedia.Siren.Parse(nonEditableCategories);
+		editableEntity = window.D2L.Hypermedia.Siren.Parse(editableCategories);
+		emptyEntity = window.D2L.Hypermedia.Siren.Parse(emptyCategories);
+	});
+
+	afterEach(() => {
+		fetchMock.reset();
+	});
+
+	describe('Categories', () => {
+		it('GetsCategories', () => {
+			var categoriesEntity = new CategoriesEntity(editableEntity);
+			const categories = categoriesEntity.getCategories();
+
+			expect(Array.isArray(categories)).to.be.true;
+			expect(categories.length).to.equal(3);
+
+			const { stringify } = JSON;
+			expect(stringify(categories[0])).to.equal(stringify(expectedCategory));
+		});
+
+		it('GetsCategoriesWhenNonePresent', () => {
+			var categoriesEntity = new CategoriesEntity(emptyEntity);
+			const categories = categoriesEntity.getCategories();
+
+			expect(Array.isArray(categories)).to.be.true;
+			expect(categories.length).to.equal(0);
+		});
+
+		it('GetsSelectedCategoriesWhenPresent', () => {
+			var categoriesEntity = new CategoriesEntity(editableEntity);
+
+			const selected = categoriesEntity.getSelectedCategory();
+
+			expect(selected.properties.name).to.equal(expectedCategory.name);
+			expect(selected.class.includes('selected')).to.be.true;
+		});
+
+		it('GetsSelectedCategoriesWhenNotPresent', () => {
+			var categoriesEntity = new CategoriesEntity(nonEditableEntity);
+			const selected = categoriesEntity.getSelectedCategory();
+
+			expect(selected).to.be.undefined;
+		});
+
+		it('GetsSelectedCategoriesWithNoActions', () => {
+			var categoriesEntity = new CategoriesEntity(nonEditableEntity);
+			const selected = categoriesEntity.getSelectedCategory();
+
+			expect(selected).to.be.undefined;
```

Comment:
Can't a non-editable entity still have a selected class? I.e. should the UI be able to see that an assignment has a category (if it does) even if we can't modify it?